modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/workflow/TenantCreationExternalWorkflowExecutor.java

```diff
@@ -0,0 +1,178 @@
+                    + "empty for the tenant domain : " + workflowDTO.getTenantDomain();
+            throw new AppFactoryException(message);
+        }
+
+        String value = createPayload(tenantCreationWorkflow);
+        ServiceClient serviceClient = null;
+        try {
+
+            ConfigurationContext context = ServiceHolder.getInstance().getConfigContextService()
+                    .getClientConfigContext();
+            serviceClient = new ServiceClient(context, null);
+            serviceClient.engageModule(WorkflowConstant.MODULE_RAMPART);
+            serviceClient.engageModule(WorkflowConstant.MODULE_ADDRESSING);
+
+            Options options = serviceClient.getOptions();
+            options.setTo(new EndpointReference(EPR));
+            options.setAction(WorkflowConstant.BPS_ACTION_URI);
+
+            Policy policy = loadPolicy(CarbonUtils.getCarbonConfigDirPath() + WorkflowConstant.BPEL_POLICY_FILE_PATH);
+
+            options.setProperty(RampartMessageData.KEY_RAMPART_POLICY, policy);
+
+            options.setUserName(AppFactoryUtil.getAppfactoryConfiguration()
+                    .getFirstProperty(AppFactoryConstants.SERVER_ADMIN_NAME));
+            options.setPassword(AppFactoryUtil.getAppfactoryConfiguration()
+                    .getFirstProperty(AppFactoryConstants.SERVER_ADMIN_PASSWORD));
+            OMElement payload = AXIOMUtil.stringToOM(value);
+            serviceClient.sendReceive(payload);
+
+        } catch (AxisFault axisFault) {
+            String message = "Unable to do the service class to the BPS server";
+            throw new AppFactoryException(message, axisFault);
+        } catch (FileNotFoundException e) {
+            String message = "Unable to find the bpel-policy.xml file";
+            throw new AppFactoryException(message);
+        } catch (XMLStreamException e) {
+            String message = "Unable to parse the xml";
+            throw new AppFactoryException(message, e);
+        } finally {
+            closeServiceClient(serviceClient);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("The BPEL ran successfully to create tenant in the cloud. Tenant domain is : " +
+                    tenantCreationWorkflow.getTenantDomain() + ". Tenant Id is : " + tenantCreationWorkflow
+                    .getTenantId());
+        }
+    }
+
+    private void closeServiceClient(ServiceClient serviceClient) {
+        try {
+            if (serviceClient != null) {
+                serviceClient.cleanup();
+            }
+        } catch (AxisFault axisFault) {
+            String message = "Unable to clean up the resources which is used by the client";
+            log.error(message, axisFault);
+        }
+    }
+
+    private String createPayload(TenantCreationWorkflowDTO tenantCreationWorkflow) {
+        String value = "<p:CreateTenantRequest xmlns:p=\"http://wso2.org/bps/sample\">" +
```

Comment:
Strings are immutable. Better using string builder if possible. By looking at it looks like a constant to me. WDYT?